### DISCUSSION TITLE
Accelerated drag scroll

### DISCUSSION THREAD

**Discussion by Valexr** [Valexr/Slidy#7] [Ideas]
Any idea about accelerate drag scroll?

**Comment by Pho3nixHun** [comment]
I suppose it needs a some kind of physics, but I'll take a look around the net, because this seems like something someone already implemented 😄

**Comment by Pho3nixHun** [comment]
These looks like something you want: 
* https://www.cssscript.com/demo/momentum-scrolling-smooth/
* https://www.cssscript.com/demo/smooth-momentum-scrolling-butter/

(Try it out in device emulation via chrome dev tools)

**Comment by Valexr** [comment]
So so... I want repeat this behaviour:
https://codepen.io/valexr/pen/xxVXEPM

**Comment by Valexr** [comment]
https://user-images.githubusercontent.com/5656366/102720236-592d4000-4304-11eb-8436-3cbb8bf4d806.mp4

**Comment by Pho3nixHun** [comment]
This does not seems to be soo hard. I mean it is a time sink -until you figure out the correct physics-, but should be relatively strait forward.

First I would calculate a ```dragForce``` from ```dragStart``` and ```dragEnd``` positions, then I would do a quick calculation which is the last image this force can drag into the center adjust ```dragForce``` to exactly bring the image into the center (or this could be an optional feature to always end on an image in center).
If during animation the user drags the image again, I would just add the new ```dragForce``` to the current one.

**Reply by Valexr** [reply]
dragForce - it's distance?

**Reply by Pho3nixHun** [reply]
I would imagined it is ```Δdistance / Δtime``` like in physics.
(```Δ``` = delta - means change - no offense meant, just in case someone not on the same level with physics or it differs between cultures)
```Δtime``` = timespan while the drag took
```Δdistance``` = distance between ```dragStart``` and ```dragEnd```

ps.: Wait we need deceleration, not speed :)

**Reply by Valexr** [reply]
ok dragSpeed... but how it implemented ? position * dragSpeed?

**Reply by Valexr** [reply]
no, we need acceleration )

**Reply by Valexr** [reply]
more precisely we need + to distance & + to transition speed from acceleration of drag

**Reply by Pho3nixHun** [reply]
So we need initial speed which is 
<img src="http://latex.codecogs.com/svg.latex?v_{initial}&space;=&space;\Delta&space;distance&space;/&space;\Delta&space;time" title="http://latex.codecogs.com/svg.latex?v_{initial} = \Delta distance / \Delta time" /> 
(```Δdistance``` and ```Δtime``` we calculate based on the interaction)

We need deceleration which is 
<img src="http://latex.codecogs.com/svg.latex?a_{deceleration}&space;=&space;(0-\Delta&space;v)&space;/&space;\Delta&space;t_{decelaration}" title="http://latex.codecogs.com/svg.latex?F(a_{deceleration}) = (0-\Delta v) / \Delta t_{d}" />
(```Δtd``` is deceleration time, could be the same or say like 80% of ```Δt``` from the previous calculation)

So from all of this we need the distance at given time, which should look something like this:
<img src="http://latex.codecogs.com/svg.latex?F_{distance}(t)&space;=>&space;v_{initial}*&space;(-(1&space;-&space;t_{deceleration})&space;*&space;t)^2" title="http://latex.codecogs.com/svg.latex?F_{distance}(t) => v_{initial}* (-(1 - t_{deceleration}) * t)^2" />

In case user does something during animation we need to calculate a new initial speed and add it to the equation.
With animation frames the rest should be good, but my brain is dead for today, so I might be wrong with the math part. (I'm a bit rusty)

**Reply by Valexr** [reply]
O! Cool calculation👍🏻Thnx!

**Reply by Valexr** [reply]
I’m rusty too )))

**Reply by Valexr** [reply]
You nickname it’s about elixir Phoenix?

**Reply by Pho3nixHun** [reply]
Nope, It just popped out of my head. I just liked the idea, that this creature can start over from any situation. :)

**Comment by Valexr** [comment]
if you have touchpad swipe it on Slidy & you'll see behaviour what I want from drag

**Reply by Pho3nixHun** [reply]
I don't have my laptop near me right now. I'll check tomorrow.

**Reply by Valexr** [reply]
Okey! I’ll wait feedback from you 👌🏻

**Reply by Valexr** [reply]
You have MacBook or..? Touchpads may be differents behaviour... correct on MacBook exactly 👍🏻

**Reply by Pho3nixHun** [reply]
I have a surface book 3 and it does work the same way (similar like in your video) on it. On my current project it does not, but probably this is some option I did not enabled.

**Reply by Valexr** [reply]
Can i see yours integration of Slidy?

**Reply by Valexr** [reply]
Thnx for access to repo, I think you have some online demo🤓

**Comment by Valexr** [comment]
Okey. Can you look for a bug with many drag swipes?

**Reply by Pho3nixHun** [reply]
I don't have any bug with it. I tried it with the double finger swipe, with touch and with mouse, but it works fine on https://valexr.github.io/slidy-site/

**Comment by Valexr** [comment]
Hey @Pho3nixHun ✌🏻
in 3.0 version release accelerated & inertial scroll -  https://svelte.dev/repl/de699aa1f8c04874b0402352ac93df96
